index.js

```diff
@@ -6,6 +6,51 @@ const KEYCODE = {
 }
 
 const state = new Map()
+// when container or children get focus
+const onFocusin = e => {
+  const {target: rover} = e
+  if (state.get('last_rover') == rover) return
+  if (state.has(rover)) {
+    activate(rover, state.get(rover).active)
+    state.set('last_rover', rover)
+  }
+}
+const onKeydown = e => {
+  const {target: rover} = e
+  switch (e.keyCode) {
+    case KEYCODE[isRtl ? 'LEFT' : 'RIGHT']:
+    case KEYCODE.DOWN:
+      e.preventDefault()
+      focusNextItem(rover)
+      break
+    case KEYCODE[isRtl ? 'RIGHT' : 'LEFT']:
+    case KEYCODE.UP:
+      e.preventDefault()
+      focusPreviousItem(rover)
+      break
+  }
+}
+const mo = new MutationObserver((mutationList, observer) =>{
+  mutationList.forEach(mutation => {
+     if(mutation.removedNodes.length > 0){
+       mutation.removedNodes.forEach(e => {
+         if (state.has(e)) {
```

Comment:
`e` here works only if the deleted node is exactly the rovingIndex container, not if it's a child of e. may need to cross reference state targets as possible children of the mutated node. 

i was able to test that here in this debug demo https://cdpn.io/pen/debug/rNpmyXL. delete the parent section and not the media-scroller, see that `e` is the section and wont clean up the events or state because of it.

Reply:
Yeah my actual use case was too simple for heavy testing. I was trying to replicate the DOMNodeRemovedFromDocument functionality the best I could.